lib/peer.ml

```diff
@@ -1,26 +1,27 @@
-type state =
+open Bin_prot.Std
+
+type status =
   | Alive
   | Suspicious
   | Faulty
 
-type t = {
+type address = {
```

Comment:
I agree with the creation of the address type so we can pack/unpack the address and send it around, but I think it makes for a cleaner API if this type isn't exposed and instead of other functions accepting `Peer.address`, they just accept `Peer.t` and use some function to access the address as necessary or construct the peer from an address.